### PR TITLE
Added limit of the failed cookie-sync messages (close #1193)

### LINKED ISSUE #1193: Add limit of the failed cookie-sync messages

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @AlexanderMoskovkin @churkin

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/cookie/cookie-sync.js [hunk 1]
```diff
@@ -5,13 +5,15 @@ import { stringify as stringifyJSON } from '../../json';
 import { isIE9 } from '../../utils/browser';
 
 const SWITCH_BACK_TO_ASYNC_XHR_DELAY = 2000;
+const FAILED_MESSAGES_LIMIT          = 3;
```
Comment (miherlosev) [inline src/client/sandbox/cookie/cookie-sync.js]: `FAILED_REQUESTS_LIMIT`

File: src/client/sandbox/cookie/cookie-sync.js [hunk 2]
```diff
@@ -5,13 +5,15 @@ import { stringify as stringifyJSON } from '../../json';
...
 
 export default class CookieSync {
     constructor () {
         this.useAsyncXhr = true;
         this.queue       = [];
         this.sendedQueue = null;
         this.activeReq   = null;
+        this.failsCount  = 0;
```
Comment (miherlosev) [inline src/client/sandbox/cookie/cookie-sync.js]: `failedReqCount`

File: test/client/fixtures/sandbox/cookie-test.js [hunk 3]
```diff
@@ -234,3 +194,23 @@ if (!browserUtils.isIOS) {
         nextCookieTest(0);
     });
 }
+
+asyncTest('limit of the failed cookie-sync messages (GH-1193)', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/cookie-test.js]: `limit of the failed cookie-sync requests (GH-1193)`

File: test/client/fixtures/sandbox/cookie-test.js [hunk 4]
```diff
@@ -234,3 +208,21 @@ if (!browserUtils.isIOS) {
...
+    var storedCookieSyncUrl = settings.get().cookieSyncUrl;
+
+    settings.get().cookieSyncUrl = '/cookie-sync-fail/';
+
+    setCookie('a=b');
+
+    setTimeout(function () {
+        strictEqual(cookieSync.failsCount, 3);
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/cookie-test.js]: Also need to override the `_onRequestError` and check that this is called 3 or more times

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Pull request merged**